### PR TITLE
Handle update of fog-ros-baseimage

### PR DESCRIPTION
**Pull request by maseabunikie**

### CODE REVIEW

**Review by joonas-fi — APPROVED** [review]

File: .github/workflows/tii-mavlink-router.yaml [hunk 1]
```diff
@@ -22,6 +24,18 @@ jobs:
 
       - uses: docker/setup-buildx-action@v1
 
+      - name: Set image tag format without suffix
+        run: |
+          echo "IMAGE_TAG_FORMAT=type=sha" >> $GITHUB_ENV
+        if: github.event_name == 'push'
```
Comment (mehmetkillioglu) [inline .github/workflows/tii-mavlink-router.yaml:30]: I think this prevents pushing the images for the PRs. Do we want to build the PRs as well, I think its useful for quick testing. Currently, there are no containers published for this PR.

Comment (joonas-fi) [reply .github/workflows/tii-mavlink-router.yaml:30]: Either of these get executed for every workflow run:

image

It's like:

```go
if {
  format = when_push
} else {
  format = when_repo_dispatch
}
```

Comment (mehmetkillioglu) [reply .github/workflows/tii-mavlink-router.yaml:30]: Yes, but it does not include `github.event_name=pull_request` still. If you check the packages, there are no images for this PR

Comment (joonas-fi) [reply .github/workflows/tii-mavlink-router.yaml:30]: Ah true, sorry I got confused when the comment was in this line.. yeah it's true this commit doesn't have pushed images but this `if` by itself doesn't restrict the pushing. You highlighted a problem however, there's still strict workflow and push restrictions in effect:

- https://github.com/tiiuae/mavlink-router/blob/3a255eb1c85cc4176dfa6e77980fde17b92e2781/.github/workflows/tii-mavlink-router.yaml#L5
- and https://github.com/tiiuae/mavlink-router/blob/3a255eb1c85cc4176dfa6e77980fde17b92e2781/.github/workflows/tii-mavlink-router.yaml#L48

Those are probably the main culprits, but you might've highlighted additional problem. Should the "without suffix" be `event_name != "repository_dispatch"` to include pull_request events as well?

File: .github/workflows/tii-mavlink-router.yaml [hunk 2]
```diff
@@ -30,8 +44,8 @@ jobs:
           tags: |
             type=ref,event=branch
             type=semver,pattern={{version}}
-            type=sha
             type=raw,value=latest
+            ${{ env.IMAGE_TAG_FORMAT }}
 
       - name: Login to GitHub Container Registry
```
Comment (mehmetkillioglu) [inline .github/workflows/tii-mavlink-router.yaml:50]: Okay, the problem was actually in the `if: github.event_name == 'push'` for Github container registry. The other PRs don't have this if condition, therefore the PRs are also built and pushed to the GHCR. Any special case why we don't want mavlink-router to push image for PRs?

Comment (mehmetkillioglu) [reply .github/workflows/tii-mavlink-router.yaml:50]: Wrong code snippet showing here. I meant Line 52.

Comment (joonas-fi) [reply .github/workflows/tii-mavlink-router.yaml:50]: Oh yeah the problem is in at least three places it seems:
- the workflow filters
- this login filter
- the push filter

Those should all be taken away to be in line with the other projects

Comment (joonas-fi) [reply .github/workflows/tii-mavlink-router.yaml:50]: > Any special case why we don't want mavlink-router to push image for PRs?

I think it's just a relic of the past. Other repos got "modernized" (= workflow filters removed) but this didn't yet receive the same love

Comment (maseabunikie) [reply .github/workflows/tii-mavlink-router.yaml:50]: I don't see the reason why not to have PRs built and pushed to GHCR.
I will remove this line.

**Pull request merged**